src/datapump/jobs/geotrellis.py

```diff
@@ -803,7 +829,10 @@ def _applications() -> List[Dict[str, str]]:
         ]
 
     @staticmethod
-    def _configurations() -> List[Dict[str, Any]]:
+    def _configurations(worker_count: str) -> List[Dict[str, Any]]:
+        executor_count = worker_count * 7
+        partition_count = executor_count * 3
```

Comment:
I've generally understood that Spark operates best when `executor_count = ((num_nodes * num_cores_per_node) / 5) - 1`, and `spark.executor.cores` = 5.  The main reason for this is that if a given executor is blocked on I/O, it can use another core for parallel processing, and 5 is typically thrown around as ideal since there's a lot of IO going on with spark (regardless of whether you're using S3 or HDFS).  The `- 1` part ensures that there will be one less executor on the node running the driver (which I think is one of the Core nodes).  The `paralellism` and `partitions` settings don't have any clear guidelines unfortunately, typically one has to try various settings and see what performs best.  I also don't know how well this applies to GeoTrellis but just wanted to give some recommendations, you may have already tried other settings!

Reply:
Oh that's interesting, I've never played around with having multiple cores for the drivers, but that makes sense. I might need to think that over and do some experiments, because right now each node is an 8 core machine, so I'm not sure if there'd be some weirdness executors spread across multiple nodes. But yeah, so far we've kept simple with just one executor per core.

For parallelism and partitions, I actually just figured out a good optimization that I'm adding in this PR. Previously, we were using 10 times the number of cores for partitions, but that ended up creating many very small partitions. Big partitions tended to run into memory issues, so I think we just slowly kept increasing it over time. 

But our major operation is a mapPartitions that re-uses S3 data across the partition, so small partitions were slowing things down by requiring a lot more reads. So I played around and found that about 3 times the number of cores made big enough partitions that it took better advantage of this, but not so big that it caused memory issues.

Reply:
Yeah, with 8 core machines it might make sense to do 3 cores per executor, which would be 2 executors per node and leave 2 cores for the OS, which is pretty reasonable.  It could be worth considering 16 core machines, which would allow 3 executors with 5 cores each, and 1 left over for the OS.  Executors can't actually be spread across nodes, so it would be a best effort to try and get as many to fit as you can, while leaving some resources for the OS.  I've seen pretty significant speed-ups from getting this ratio right.

Re: partitions, that all sounds right to me, basically you want to take into account the size of the data and the overall compute resources, and come up with a good compromise.  Too small = too much IO as you said, too big results in expensive shuffle operations if data needs to be moved between nodes.

Reply:
Yeah, 16 cores seems like a good idea then. I'll create a separate jira ticket to do some performance testing with that.